src/main/scala/com/amazon/deequ/analyzers/MaxLength.scala

```diff
@@ -35,12 +36,12 @@ case class MaxLength(column: String, where: Option[String] = None, analyzerOptio
   with FilterableAnalyzer {
 
   override def aggregationFunctions(): Seq[Column] = {
-    max(criterion) :: Nil
+    max(element_at(criterion, 2).cast(DoubleType)) :: Nil
```

Comment:
nit: could we add a comment here explaining this behavior? Think it would be beneficial to clarify that this criterion is returning a column of list/tuples.